pom.xml

```diff
@@ -151,64 +128,55 @@
               <instanceId>bitbucket</instanceId>
-              <version>${bitbucket.version}</version>
-              <dataVersion>${bitbucket.data.version}</dataVersion>
+              <version>${bitbucket.test.version}</version>
+              <dataVersion>${bitbucket.test.version}</dataVersion>
             </product>
           </products>
+          <testGroups>
+            <testGroup>
+              <id>bitbucket-integration</id>
+              <includes>
+                <include>it/**/*.java</include>
+              </includes>
+              <productIds>
+                <productId>bitbucket</productId>
+              </productIds>
+            </testGroup>
+          </testGroups>
         </configuration>
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.3</version>
+        <version>3.7.0</version>
         <configuration>
           <source>1.8</source>
           <target>1.8</target>
         </configuration>
       </plugin>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>build-helper-maven-plugin</artifactId>
-        <version>3.0.0</version>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.20.1</version>
+        <configuration>
+            <excludes>
+              <exclude>it/**/*.java</exclude>
+            </excludes>
+        </configuration>
       </plugin>
     </plugins>
   </build>
 
-  <profiles>
-    <profile>
-      <id>integration</id>
-      <activation>
-        <activeByDefault>false</activeByDefault>
-      </activation>
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>org.codehaus.mojo</groupId>
-            <artifactId>build-helper-maven-plugin</artifactId>
-            <executions>
-              <execution>
-                <phase>generate-sources</phase>
-                <goals>
-                  <goal>add-test-source</goal>
-                </goals>
-                <configuration>
-                  <sources>
-                    <source>${project.basedir}/src/test/java/it</source>
-                  </sources>
-                </configuration>
-              </execution>
-            </executions>
-          </plugin>
-        </plugins>
-      </build>
-    </profile>
-  </profiles>
-
   <properties>
-    <bitbucket.version>5.0.1</bitbucket.version>
-    <bitbucket.data.version>5.0.1</bitbucket.data.version>
-    <amps.version>6.3.0</amps.version>
+    <bitbucket.version>4.8.6</bitbucket.version>
+    <bitbucket.data.version>${bitbucket.version}</bitbucket.data.version>
+    <!-- This separate version allows testing the plugin with a different version of the system than
+         it's compiled against, which is useful for testing compatibility with multiple versions -->
+    <bitbucket.test.version>${bitbucket.version}</bitbucket.test.version>
```

Comment:
I'd introduced this separate version to allow, for example, compiling PR Harmony against Bitbucket Server 4.8 (which is the lower bound of the compatibility range you've set on Marketplace), but running the tests against Bitbucket Server 5.8 (the upper bound).

You can run the integration tests against a given version with `mvn clean install -DskipITs=false -Dbitbucket.test.version=5.8.1`

Reply:
Nice!